src/embind/embind.js

```diff
@@ -1131,7 +1131,7 @@ var LibraryEmbind = {
     invokerFnBody += "return Asyncify.currData ? Asyncify.whenDone().then(onDone) : onDone(" + (returns ? "rv" : "") +");\n"
 #elif ASYNCIFY == 2
     invokerFnBody += "}\n";
-    invokerFnBody += "return " + (isAsync ? "rv.then(onDone)" : "onDone(rv)") + ";";
+    invokerFnBody += `return ${isAsync ? "rv.then(onDone)" : `onDone(${returns ? "rv" : ""})`};`;
```

Comment:
Sadly we haven't starting allowing JS template literals in our generated code yet.   We allow certain ES6 features such as arrow functions that can easily be lowered away by closure compiler, but we don't yet allow this one.

Looking at  https://caniuse.com/template-literals, it seems like perhaps we should allow it though!  

In any case, we should probably not introduce it here and now, but in a more principled fashion.

Reply:
Sure thing, will use the good old string concatenation.

Reply:
Also, would be happy to help with introducing that.

Reply:
I took a stab at it yesterday: https://github.com/emscripten-core/emscripten/pull/19285